fem/qspace.hpp

```diff
@@ -58,6 +62,15 @@ class QuadratureSpaceBase
    /// Compute the integration weights.
    void ConstructWeights() const;
 
+   /// @brief Gets the offset for a given entity @a idx.
+   ///
+   /// The quadrature point values for entity i are stored in the indices
+   /// between Offset(i) and Offset(i+1)
+   int Offset(int idx) const
+   {
+      return (offsets.Size() == 1) ? (idx * offsets[0]) : offsets[idx];
+   }
```

Comment:
While you're adding this addition to the `QuadratureSpaceBase` class, it would probably also be useful to add a getter function like:

```c++
const Array<int>& getOffsets() const { return offsets; }
```

Since, it can be useful to directly use the offset arrays in GPU-compatible kernels that don't reside within the `QuadratureFunction` class. I've got a few use cases where this would make my life much simpler for the GPU-compatible kernels I'm writing. It could also allow future versions of the `QuadratureInterpolator` to be usable with non-uniform strides for the integration points across the mesh as well.